tests/page-objects/collection-facets.ts

```diff
@@ -48,57 +60,65 @@ export class CollectionFacets {
 
     const datePicker = await this.getFacetGroupContainer(
       FacetGroupLocatorLabel.DATE,
+      FacetGroupFilterHeaderEnum.YEAR,
     );
     expect(await datePicker?.innerText()).toContain('Year');
   }
 
   async selectFacetByGroup(
     group: FacetGroupLocatorLabel,
+    header: FacetGroupFilterHeaderEnum,
     facetLabel: string,
     facetType: FacetType,
   ) {
-    await this.page.waitForLoadState('networkidle', { timeout: 60000 });
-
-    const facetContent = await this.getFacetGroupContainer(group);
-    if (facetContent) {
-      if (facetType === 'positive') {
-        const facetRow = facetContent.getByRole('checkbox', {
-          name: facetLabel,
-        });
-        await facetRow.check();
-        return;
-      }
-
-      if (facetType === 'negative') {
-        const facetRows = await facetContent
-          .locator('facets-template > div.facets-on-page facet-row')
-          .all();
-        if (facetRows.length !== 0) {
-          for (let x = 0; x < facetRows.length; x++) {
-            const facetRowLabel = facetRows[x].locator(
-              'div.facet-row-container > div.facet-checkboxes > label',
-            );
-            const facetRowTitle = await facetRowLabel.getAttribute('title');
-            if (facetRowTitle === `Hide mediatype: ${facetLabel}`) {
-              await facetRowLabel.click();
-              return;
-            }
+    const facetContent = await this.getFacetGroupContainer(group, header);
+    const facetContentInnerHTML = await facetContent?.innerHTML();
+
+    if (facetContentInnerHTML?.includes('facets-template')) {
+      const facetRows = await facetContent
+        ?.locator('facets-template > div.facets-on-page facet-row')
+        .all();
+      if (facetRows && facetRows.length !== 0) {
+        for (let x = 0; x < facetRows.length; x++) {
```

Comment:
SUGGESTION: Since you don't need the value of `x` other than for iteration of the `facetRows`, while not use a for..of loop?